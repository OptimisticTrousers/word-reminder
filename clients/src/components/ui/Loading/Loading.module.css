.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
}

.loading__image {
  max-inline-size: 100%;
}
