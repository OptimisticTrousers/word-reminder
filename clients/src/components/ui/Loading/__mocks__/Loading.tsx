export function Loading() {
  return <div data-testid="loading">Loading...</div>;
}
