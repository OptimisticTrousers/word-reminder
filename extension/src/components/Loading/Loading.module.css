.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.loading__image {
  max-inline-size: 100%;
}
