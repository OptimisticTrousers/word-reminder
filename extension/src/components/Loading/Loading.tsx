import CSSModules from "react-css-modules";
import styles from "./Loading.module.css";

const Loading = CSSModules(
  () => {
    return (
      <div styleName="loading">
        <img
          styleName="loading__image"
          src={"/images/loading.gif"}
          alt="loader"
        />
      </div>
    );
  },
  styles,
  {
    allowMultiple: true,
    handleNotFoundStyleName: "ignore",
  }
);

export default Loading;
