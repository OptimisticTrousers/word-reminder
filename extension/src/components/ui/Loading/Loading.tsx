import CSSModules from "react-css-modules";

import loadingImage from "../../../assets/images/loading.gif";
import styles from "./Loading.module.css";

export const Loading = CSSModules(
  function () {
    return (
      <div styleName="loading">
        <img styleName="loading__image" src={loadingImage} alt="loading" />
      </div>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
