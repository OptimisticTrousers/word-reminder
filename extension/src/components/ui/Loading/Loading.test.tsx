import { render, screen } from "@testing-library/react";

import loadingImage from "../../../assets/images/loading.gif";
import { Loading } from "./Loading";

describe("Loading component", () => {
  it("renders image", () => {
    const { asFragment } = render(<Loading />);

    const image = screen.getByAltText("loading");

    expect(image).toHaveAttribute("src", loadingImage);
    expect(asFragment()).toMatchSnapshot();
  });
});
