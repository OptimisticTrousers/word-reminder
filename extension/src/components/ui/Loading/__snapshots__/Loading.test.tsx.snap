// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading component > renders image 1`] = `[Function]`;
